src/lib/models/transaction.js

```diff
@@ -309,6 +309,78 @@ class Transaction extends Model {
     return this._isDeMinimis;
   }
 
+  set calcSourceFactor (calcSourceFactor) {
+    validators.assertNullableNumeric(calcSourceFactor);
+    this._calcSourceFactor = calcSourceFactor;
+  }
+
+  get calcSourceFactor () {
+    return this._calcSourceFactor;
+  }
+
+  set calcSeasonFactor (calcSeasonFactor) {
+    validators.assertNullableNumeric(calcSeasonFactor);
+    this._calcSeasonFactor = calcSeasonFactor;
+  }
+
+  get calcSeasonFactor () {
+    return this._calcSeasonFactor;
+  }
+
+  set calcLossFactor (calcLossFactor) {
+    validators.assertNullableNumeric(calcLossFactor);
+    this._calcLossFactor = calcLossFactor;
+  }
+
+  get calcLossFactor () {
+    return this._calcLossFactor;
+  }
+
+  set calcSucFactor (calcSucFactor) {
+    validators.assertNullableNumeric(calcSucFactor);
+    this._calcSucFactor = calcSucFactor;
+  }
+
+  get calcSucFactor () {
+    return this._calcSucFactor;
+  }
+
+  set calcS126Factor (calcS126Factor) {
```

Comment:
For the S126 and S127 factors, could we map these to a number and use a basic numeric type?